### DISCUSSION TITLE
ELRS Receiver binding procedure is a pain for newcomers.

### DISCUSSION THREAD

**Discussion by Alt-Pavlo-Delete** [ExpressLRS/ExpressLRS#2017] [General]
Greetings from Ukraine! 
I just want to leave feedback about current binding procedure. 
I bought near 20 different ELRS 900RX receivers from JHEMCU, Happymodel, BetaFPV.
Fistly it was huge surprise for me that`s familiar to everyone way to enter binding mode with a hardware button does not works on ELRS. 

25% receivers refused to enter binding mode after 3 power cycles, so i was forced to waste tonn of my time to reserch all ELRS peculiarities. I had to find how this "wonderful" binding phrase can be changed, solder TTL converters, download configurator, read manuals, etc, etc. And user forced to do all that insted just: "hold button and power on", why? It messes up all user experience.

Is it possible to use hardware button to enter binding mode? 
Why good old, familiar to everyone binding procedure with a harware button was not ued?
Using the same procedure as all manufacturers on the market would be a significant improvement for users (especially newcomers).

**Comment by commanderguy3001** [comment]
generally elrs is designed around the idea that a user should flash their receiver with the most recent version of elrs once they install it, and that people are able to provide a binding phrase at that stage too, wich removes the hassle of manual binding later.

I really don't understand why you messed with TTL converter either, in most cases you can use betaflight passthrough just fine to flash your receivers.

regarding newcomers (to the hobby), no, it is NOT easier for them to understand traditional binding, as they don't know anything yet and will have to learn one version anyway.
And if you have to learn something anyway, you should go with bind phrases, as they objectively are the better way of binding (you don't have to agree to this, but it is true either way, make an objective list of pros and cons and you'll see).

The reason why it got introduced at all: innovation, as that's one of the main things that elrs is about. Someone at some point just didn't want to mess with old style binding anymore, which is especially tedious since you would have to do it every time again after flashing a new version of elrs. so, someone implemented bind phrases which were just a way to skip that clumsy old manual binding process by just entering some text while flashing the receiver, which was/is done pretty frequently, as elrs development is going really fast.

**Comment by commanderguy3001** [comment]
Also, generally, I'd recommend you read the [official documentation here](https://expresslrs.org/3.0/quick-start/getting-started). You probably would have saved yourself some confusion when you would have looked at that before trying to start using elrs.

**Reply by Alt-Pavlo-Delete** [reply]
You wont believe, but first of all i read original documentation.
Everything in this thread is just my personal opinion, just opinion of the random user who switched from Frsky, Tbs, and etc. And i am sure that i am not alone. Of course you can say thats i am free to use any other solution if i didn't like this one (ELRS). But if you interested in further product improvement user experience should be  taken into account.

So i was forced to read troubleshoot (to find thats its all about bind phrase) than forced to download and install configurator (more configurators to the god of configurators) i, than when configurator did not discovered receiver connected to the betaflight controller (i dont know why) i had to solder TTL dongle, and find some youtube videos how to change bind phrase and flash receivers. 
For sure its easier and faster than "Hold button and power on".
Maybe Frsky, Flysky, TBS, and other guys just don't get something.

**Reply by commanderguy3001** [reply]
well, this is how I experienced it:
- I heard of elrs, and went to the discord
- learned a little bit about how elrs is supposed to work
- ordered my gear
- installed the configurator when my stuff arrived, went to flashing, and it worked just fine without any problems (actually, I did run into one issue which was solved by reading a couple lines in the documentation)
- it "magically" _just worked_

**Reply by Alt-Pavlo-Delete** [reply]
Honestly, as i mentioned above, first 10+ units worked fine, it was quick, i was lucky to receive receivers flashed from the factory without binding phrase. Only 5 receivers out of 20 failed to bind with usual procedure. And required flashing. Its okay for hobby if someone wants to learn something new, read some manuals instead of flying.
